transactor/src/main/scala/io/mediachain/transactor/StateMachine.scala

```diff
@@ -111,6 +124,27 @@ object StateMachine {
         commit.release()
       }
     }
+    
+    def currentBlock(commit: Commit[JournalCurrentBlock]) : JournalBlock = {
+      try {
+        JournalBlock(seqno, blockchain, block.toArray)
+      } finally {
+        commit.release()
+      }
+    }
+
+    // block generation
+    private def blockExtend(entry: JournalEntry) {
+      block += entry
+      if (block.length >= blocksize) {
```

Comment:
I still have to release the commit and return the value, so I adopted this pattern to have visual symmetry with the other API entry points.

Alternative is to write it this way:
```scala
val block = JournalBlock(...)
commit.release()
block
```

which I find uglier (and also incogruous with the other API methods)